### PR TITLE
shows search component tested for

### PR DESCRIPTION
**Pull request by JacobMacFarlane**
____ Wrote Tests ____ Implemented ____ Reviewed

## Neccesary checkmarks
- [ ] All Tests are Passing in all environments
- [ ] The code will run locally
## Type of change
- [X ] New feature
- [ ] Bug Fix
- [ ] Refactor
## Description of Change:

  description:  I added testing for the following flows
  If a user can succesfully search for what they are looking for
  If a user can favorite a card(or 2)
  If a user can view whatever card they favorite through the favorite filter
  If a user can unfavorite within favorite and hit clear and that item is no longer "fav.png"
  If a user can favorite click the favorite and then hit clear and see both cards
  If a user can view the favorite filter and see no favorites by default
    
## Requested feedback: 
    
  I'd like feedback on how to view the different pngs, i could not figure it out but i think i was specific enough
  If i was not specific enough in the tests please let me know

## Check the correct boxes

- [X ] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes

- [ ] No Tests have been changed
- [X ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)

## Checklist:

- [ ] My code has no unused/commented out code
- [ ] My code has no binding.pry's
- [ ] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X ] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link:

### CODE REVIEW

File: cypress/e2e/search_spec.cy.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    cy.get('.downvotes').contains('p', '12')
+  })
+  it('Should be able to find the favorite button', () => {
+    cy.get(`[value="favorites"]`).should('exist')
+  })
+  it('Should have no favorites by default', () => {
+    cy.get(`[value="favorites"]`).click()
+    cy.get(`.card-container`).should('not.exist')
+  })
+  it("Should be able to favorite a card", () => {
```
Comment (Devynnem) [inline cypress/e2e/search_spec.cy.js:30]: I still haven't used nth-child, i'll have to ask you about this. But looks great!

**Review by Devynnem — APPROVED** [review]

File: cypress/e2e/search_spec.cy.js (hunk 1 above)
Comment (JacobMacFarlane) [reply cypress/e2e/search_spec.cy.js:30]: Ya so on the cypress page if you click the button on the top left that looks like a target, you can use it to click and view the exact term you need to target anything

Comment (Pma913) [reply cypress/e2e/search_spec.cy.js:30]: I have not tried that technique yet either

Comment (JacobMacFarlane) [reply cypress/e2e/search_spec.cy.js:30]: Its the method if you guys want to hop on zoom rq i can show you, itll take like a minute

**Comment by Pma913** [comment]
Tests look great. Nice work!

**Review by Pma913 — APPROVED** [review]

**Pull request merged**